rs/execution_environment/src/execution_environment.rs

```diff
@@ -418,6 +421,7 @@ impl ExecutionEnvironment {
             Arc::clone(&ingress_history_writer),
             fd_factory,
         );
+        let deallocator_thread = DeallocatorThread::new("ExecutionDeallocator", 10000);
```

Comment:
You hit the nail on the head. (o:

The State Manager deallocator thread would originally sleep for 1 ms between successive drops. So I used `1000` (as in `1 s / 1000`) as the factor there, to preserve the behavior.

I initially started with 1000 here too, but while running my (very heavy) [benchmarks](https://docs.google.com/document/d/1AX7M8SjEPNsoxxTa_FD2suT9cArU7Tw4t1f-_UIOIXU/edit?tab=t.0) I noticed that half the deallocations still happened on the scheduler threads. So I just bumped it by 10x because I wanted to get some results quickly and one benchmark run would take on the order of 15 minutes.

On the one hand, I did not immediately notice any negative effects from sleeping 100 ns between drops. And we are likely to both benefit from and need to deallocate lots of large objects under extremely heavy load. Under normal load, the 10k will neither help, nor have any effect.

That being said, I can try to fine tune it. It's quite likely that we can reduce the factor (and increase the interval) by at least 2x without negative effects on throughput. (The peak message throughput I saw in my testing was about 2.5k messages/second, so assuming 2 objects to deallocate per message, we could likely do with 5k objects sent to the deallocator thread per second, At 1 object per message, 2.5k would do.)

Reply:
I've done a couple of runs, with 3k and 5k. In both cases, some `SystemStates` and `ExecutionStates` are being dropped synchronously, because the channel to the deallocator thread becomes backlogged. With 3k (333µs sleep), it's about 20%; with 5k (200 µs sleep) it's about 13%.
![Screenshot from 2024-12-19 11-52-37](https://github.com/user-attachments/assets/9924267f-ea23-4dc7-a029-c94f8874f063)
![Screenshot from 2024-12-19 11-52-50](https://github.com/user-attachments/assets/79178ae1-ab3f-48b6-9ead-eff829e74502)
FWIW, I've never noticed any synchronous drops with 10k (100 µs sleep). But then the overhead to execution (12% and respectively 9%; only under very heavy load) is not really meaningful with either 3k or 5k.

Reply:
It needs some comment explaining why 10000 in the code and when to readjust it

Reply:
I will add a doc comment, thanks for the suggestion.

Reply:
Done.